src/components/Header/Header.tsx

```diff
@@ -29,19 +36,22 @@ const Header = (props: any): JSX.Element => {
     showModal();
   };
 
-  const handleIncrement = (): void => {
-    let incrementedSearchedTimes = searchedTimes;
-    incrementedSearchedTimes++;
-    setSearchedTimes(incrementedSearchedTimes);
-  };
+  const { data } = api.useGetMovieByIdQuery(movieQueryParameter ? movieQueryParameter : skipToken);
+
+  useEffect(() => {
+    if (data) {
+      dispatch(moviesActions.setMovie(data));
```

Comment:
I wonder if `onCacheEntryAdded` method of the rtk query can be used to set the data to the store automatically, without involving a component

Reply:
left it for now how it is